dace/frontend/python/newast.py

```diff
@@ -2194,6 +2195,15 @@ def visit_For(self, node: ast.For):
                 # The state that all "break" edges go to
                 loop_end = self._add_state(f'postloop_{node.lineno}')
 
+            body_states = set()
+            to_visit = [first_loop_state]
```

Comment:
why not use our existing traversal methods like dfs_conditional?

Reply:
Apart from being, in general, difficult to locate these methods in DaCe, the problem is that it is not always clear what they do or what their constraints are. For example, according to `dfs_conditional` documentation, the input `G` DiGraph is assumed acyclic. After looking at the code, it seems likely that the documentation is wrong, but who knows? If anything, I would have made a new method that explicitly finds the multi-state body of loops because we do that in many transformations already without using these traversal methods